modules/qualitycontrol/QualityController.ts

```diff
@@ -146,24 +155,50 @@ export default class QualityController {
 
     /**
      * Adjusts the lastN value so that fewer remote video sources are received from the bridge in an attempt to improve
-     * encode resolution of the outbound video streams.
-     * @returns boolean - Returns true if the lastN was lowered, false otherwise.
+     * encode resolution of the outbound video streams based on cpuLimited parameter passed. If cpuLimited is false,
+     * the lastN value will slowly be ramped back up to the channelLastN value set in config.js.
+     *
+     * @param {boolean} cpuLimited - whether the endpoint is cpu limited or not.
+     * @returns boolean - Returns true if an action was taken, false otherwise.
      */
-    _maybeLowerLastN(): boolean {
+    _lowerOrRaiseLastN(cpuLimited: boolean): boolean {
         const lastN = this.receiveVideoController.getLastN();
-        const videoStreamsReceived = this._conference.getForwardedSources().length;
 
-        if (lastN !== -1 && lastN <= MIN_LAST_N) {
+        // Do nothing if lastN is already set to zero, i.e., the endpoint is in audio-only mode.
```

Comment:
Hmm .. it doesn't seem to be that straight forward since audio-only feature is fully implemented in the application. However, I realized that when audio-only mode is enabled, client's local video sources are muted so the stats won't be processed and when either camera or screenshare is started, the client is automatically kicked out of audio-only mode. Therefore, we don't have to make assumptions about whether client is in that mode while calculating lastN. I have removed that check all together now.